src/connection.c

```diff
@@ -978,7 +979,8 @@ CONNECTION_HANDLE connection_create2(XIO_HANDLE xio, const char* hostname, const
 
                     if (result != NULL)
                     {
-                        result->container_id = (char*)malloc(strlen(container_id) + 1);
+                        size_t container_id_length = strlen(container_id);
+                        result->container_id = (char*)malloc(container_id_length + 1);
```

Comment:
Well, it needs to look for the end of the string and there is no need for it in this case, the length is known.